manifests/init.pp

```diff
@@ -46,47 +47,48 @@
 #   Optional string, integer or array of minute(s) the renewal command should
-#   run. E.g. 0 or '00' or [0,30]. Default: fqdn-seeded random minute.
+#   run. E.g. 0 or '00' or [0,30].
 # @param renew_cron_monthday
 #   Optional string, integer or array of monthday(s) the renewal command should
-#   run. E.g. '2-30/2' to run on even days. Default: Every day.
+#   run. E.g. '2-30/2' to run on even days.
 #
 class letsencrypt (
   Optional[String] $email                = undef,
-  String $path                           = $letsencrypt::params::path,
-  $venv_path                             = $letsencrypt::params::venv_path,
+  String $path                           = '/opt/letsencrypt',
+  $venv_path                             = '/opt/letsencrypt/.venv',
   Array $environment                     = [],
-  String $repo                           = $letsencrypt::params::repo,
-  String $version                        = $letsencrypt::params::version,
-  String $package_name                   = $letsencrypt::params::package_name,
-  $package_ensure                        = $letsencrypt::params::package_ensure,
-  String $package_command                = $letsencrypt::params::package_command,
-  String $config_file                    = $letsencrypt::params::config_file,
-  Hash $config                           = $letsencrypt::params::config,
-  String $cron_scripts_path              = $letsencrypt::params::cron_scripts_path,
-  Boolean $manage_config                 = $letsencrypt::params::manage_config,
-  Boolean $manage_install                = $letsencrypt::params::manage_install,
-  Boolean $manage_dependencies           = $letsencrypt::params::manage_dependencies,
-  Boolean $configure_epel                = $letsencrypt::params::configure_epel,
-  Enum['package', 'vcs'] $install_method = $letsencrypt::params::install_method,
-  Boolean $agree_tos                     = $letsencrypt::params::agree_tos,
-  Boolean $unsafe_registration           = $letsencrypt::params::unsafe_registration,
-  Stdlib::Unixpath $config_dir           = $letsencrypt::params::config_dir,
+  String $repo                           = 'https://github.com/certbot/certbot.git',
+  String $version                        = 'v0.30.2',
+  String $package_name                   = 'letsencrypt',
+  $package_ensure                        = 'installed',
+  String $package_command                = 'letsencrypt',
+  Stdlib::Unixpath $config_dir           = '/etc/letsencrypt',
+  String $config_file                    = "${config_dir}/cli.ini",
+  Hash $config                           = {'server' => 'https://acme-v01.api.letsencrypt.org/directory'},
+  String $cron_scripts_path              = "${facts['puppet_vardir']}/letsencrypt",
+  String $cron_owner_group               = 'root',
+  Boolean $manage_config                 = true,
+  Boolean $manage_install                = true,
+  Boolean $manage_dependencies           = true,
+  Boolean $configure_epel                = false,
```

Comment:
This is listed in data. I wonder if this should list the default `undef` here and set `false` in `common.yaml`. That way `REFERENCE.md` is indicating in a way that it's OS-dependent.

Reply:
https://travis-ci.org/voxpupuli/puppet-letsencrypt/jobs/594702592#L4142

Without `Optional` data type we are getting a message `expects a Boolean value, got Undef` during rspec tests.

Reply:
accordingly with talks on IRC we stay standard for the moment.